### PR TITLE
Changing wording and adding text helper for product name error input

### LINKED ISSUE #25905: BO - Product V2 - An exception is displayed when creating a product with invalid name or too long name
#### Describe the bug
In the BO > Catalog > Products > when we try to add a product V2 with an invalid name or too long name => an exception displayed
image

#### Solution
Capture d’écran 2022-01-27 à 12 29 24

Wording alert: 'This field contains invalid characters (<>;=#{})'

#### Steps to Reproduce

1. Go to BO > International > Taxes page > enable ecotax
2. Go to BO > Advanced Parameters > Experimental Features page
3. Enable Experimental product page
4. Go to BO > Catalog > Products page and add a product V2
5. Set the name with an invalid expression for example `<<<` or an expression too long and save
6. See error

#### Additional information

* PrestaShop version: 1.7.8.0rc1build4, develop
* PHP version: 7.2

### PR DESCRIPTION
**Pull request by lartist**
…t helper below the input

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Changing wording and adding text helper for product name error input
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #25905
| Related PRs       |
| How to test?      | https://github.com/PrestaShop/PrestaShop/issues/25905
| Possible impacts? |

### CODE REVIEW

**Review by 0x346e3730 — DISMISSED** [review]

**Review by jolelievre — DISMISSED** [review]
Thanks @lartist 

Ok for me, but you need to update the target in your PR description (to `develop`)

**Comment by lartist** [comment]
> Thanks @lartist
> 
> Ok for me, but you need to update the target in your PR description (to `develop`)

Done

**Comment by lartist** [comment]
I think a wording label is missing no?

**Review by mflasquin — DISMISSED** [review]

**Comment by jolelievre** [comment]
@lartist you used an existing wording that is adapted so there are few chances that it's incorrect. So I think this is testable, and we can wait for @l-delin confirmation and label before we merge it.

**Comment by l-delin** [comment]
Hi, you're all good to go :-)

**Comment by paulnoelcholot** [comment]
Hello @lartist,

I have tested the PR and I think that is not the expected behavior.

For a too long name I have this result : 
image

I think that is better if the wording is "This value is too long, it should have 128 characters or less. - Language : English (English)

For a invalid caracter, I have this result : 
image

The wording should be " This field contains invalid characters (<>;=#{}) "

**Comment by jolelievre** [comment]
Ah @lartist I believe you changed the constraint on the form used in V1 (which is good as well) but it's a different form type for product v2 https://github.com/PrestaShop/PrestaShop/blob/840fc123126857f93925598d47b5d86c96edf35d/src/PrestaShopBundle/Form/Admin/Sell/Product/HeaderType.php#L82

**Review by FabienPapet — DISMISSED** [review]

**Comment by lartist** [comment]
> Hello @lartist,
> 
> I have tested the PR and I think that is not the expected behavior.
> 
> For a too long name I have this result : image
> 
> I think that is better if the wording is "This value is too long, it should have 128 characters or less. - Language : English (English)
> 
> For a invalid caracter, I have this result : image
> 
> The wording should be " This field contains invalid characters (<>;=#{}) "

@Jo were right. I applied changes only in v1. Now it should be ok in v2 as well.
NB: I have taken the liberty to put the wording `This field contains invalid characters <>;=#{}` instead of `This field contains invalid characters (<>;=#{})` since the parenthesis could be confusing. Do they wrap invalid characters or are they invalid characters themselves ?

**Review by zuk3975 — APPROVED** [review]

**Review by jolelievre — APPROVED** [review]
Thanks @lartist

**Review by florine2623 — COMMENTED** [review]
Hello @lartist ,

The error message is always displayed:
Tested with all types of product, it is the same behavior.
Screenshot 2023-01-19 at 10 23 35

It should be displayed in a red alert only when the name of the product contains a forbidden character.

Could you check ? ^^

**Comment by jolelievre** [comment]
Hi @florine2623 

the message always displayed is not an error message, it's a help message indicating to the user what he shouldn't use. The error message will appear if the user tries to use these characters regardless.

**Edit** this is the solution that was decided in the issue https://github.com/PrestaShop/PrestaShop/issues/25905#issuecomment-1023330387 To have both an error message AND an appropriate helper text.

**Comment by lartist** [comment]
>Hello @lartist ,

>The error message is always displayed:
>Tested with all types of product, it is the same behavior.
>Screenshot 2023-01-19 at 10 23 35

>It should be displayed in a red alert only when the name of the product contains a forbidden character.

>Could you check ? ^^

Same than @jolelievre :)

**Review by florine2623 — APPROVED** [review]
Alright @jolelievre and @lartist ! I didn't see that comment ^^'

I do have the text helper and the error message.
No exception is displayed.

It is QA ✅

**Comment by FabienPapet** [comment]
Thank you @lartist

**Pull request merged**
